.travis.yml

```diff
@@ -1,21 +1,32 @@
+sudo: false
 language: scala
 
 jdk:
   - oraclejdk8
 
+scala:
+  - 2.10.6
+  - 2.12.0-M5
+  - 2.12.0-RC1
+  - 2.12.0-RC2
+
+matrix:
+  include:
+    - scala: 2.12.0-M4
+      env: TEST_SCRIPTED=1
+      jdk: oraclejdk8
+    - scala: 2.11.8
+      env: TEST_SCRIPTED=1
+      jdk: oraclejdk8
+
 script:
-  - sbt plugin/compile plugin/publishLocal
-  - sbt ++2.10.6     core/compile core/publishLocal
-  - sbt ++2.11.7     core/compile core/publishLocal
-  - sbt ++2.12.0-M4  core/compile core/publishLocal scripted
-  - sbt ++2.12.0-M5  core/compile core/publishLocal
-  - sbt ++2.12.0-RC1 core/compile core/publishLocal
-  - sbt ++2.12.0-RC2 core/compile core/publishLocal
+  - sbt ++$TRAVIS_SCALA_VERSION core/compile
+  - "if [[ $TEST_SCRIPTED = 1 ]]; then sbt plugin/compile plugin/publishLocal && sbt ++$TRAVIS_SCALA_VERSION core/publishLocal scripted; fi"
```

Comment:
The fact that `sbt ++2.12.0-M4 scripted` worked at all is purely accidental. `scripted` tests need to be executed with `2.10.x`. As an alternative, I'd suggest passing in the Scala version to be used _inside_ the tests via system properties. But that's an orthogonal change.